cms/djangoapps/contentstore/views/item.py

```diff
@@ -846,7 +849,14 @@ def create_xblock_info(xblock, data=None, metadata=None, include_ancestor_info=F
         "course_graders": json.dumps([grader.get('type') for grader in graders]),
         "has_changes": has_changes,
         "actions": xblock_actions,
-        "explanatory_message": explanatory_message
+        "explanatory_message": explanatory_message,
+        "enable_proctored_exams": enable_proctored_exams,
```

Comment:
I'm not comfortable with adding all of these fields to every XBlock info. The original intention of the XBlock info was that it was just the information needed to display the course outline. If I've read this correctly, these fields are not used by the outline, but are only ever used by the "Subsection Settings" editor. I don't like that every feature adds more fields to this info.

I can think of a couple of ways I'd rather do this:
1. My preferred option would be to update the edit modal to make an AJAX request to get the exam info that it needs, rather than expecting to find it in the XBlock info. I could see that this would be too big a change to make right now, though.
2. A simpler approach would be to optionally add a single optional ``exam-info`` field to XBlock info. See below for how we add ``ancestor-info`` in a similar fashion. This would then mean that most xblocks would only have one extra empty field, and it should scale in the future if we add more exam settings.

@cahrens What's your opinion?